src/main/org/apache/tools/zip/ZipOutputStream.java

```diff
@@ -901,6 +901,20 @@ public boolean canWriteEntryData(ZipEntry ae) {
         return ZipUtil.canHandleEntryData(ae);
     }
 
+    /**
+     * Writes a byte to ZIP entry.
+     *
+     * @param b the byte to write
+     * @throws IOException on error
+     * @since Ant 1.10.10
+     */
+    @Override
+    public void write(int b) throws IOException {
+        byte[] buf = new byte[1];
```

Comment:
We could allocate a one-byte array and reuse that for all calls. Of course then we'd need to think about thread safety issues. So maybe that's not wort the trouble given that so far nobody seems to have used that method anyway :-)

Reply:
Agreed, I thought about that but was wondering why `DeflaterOutputStream` didn't do it. I did it now as this class is not thread-safe anyway, and this allocation on every write may keep the GC unnecessarily busy. I checked Apache Commons Compress since you mentioned it and it's also it's done there: https://github.com/apache/commons-compress/blob/master/src/main/java/org/apache/commons/compress/archivers/ArchiveOutputStream.java#L52